tests/integration_tests/build/test_style.py

```diff
@@ -32,57 +33,78 @@ def test_rust_style():
     assert "Diff in" not in process.stdout.decode('utf-8')
 
 
+def gen_pylint_cmds(file_list):
+    """
+    Return a list of pylint commands that should be executed for each file.
+
+    :param file_list: list of files to check
+    :return: list of commands
+    """
+    base_cmd = \
+        r'python3 -m pylint --jobs=0 --persistent=no --score=no ' \
+        r'--output-format=colorized --attr-rgx="[a-z_][a-z0-9_]{1,30}$" ' \
+        r'--argument-rgx="[a-z_][a-z0-9_]{1,30}$" ' \
+        r'--variable-rgx="[a-z_][a-z0-9_]{1,30}$" --disable=' \
+        r'bad-continuation,fixme,too-many-instance-attributes,' \
+        r'too-many-locals,too-many-arguments'
+
+    cmd_list = []
+    for check_file in file_list:
+        cmd_list.append(f"{base_cmd} {check_file}")
+
+    return cmd_list
+
+
+def gen_pycodestyle_cmds():
+    """
+    Return a list of pycodestyle commands that should be executed.
+
+    :return: list of commands
+    """
+    return ['python3 -m pycodestyle --show-pep8 --show-source '
+            '--exclude=../build ../']
+
+
+def gen_pydocstyle_cmds(file_list):
+    """
+    Return a list of pydocstyle commands that should be executed for each file.
+
+    :param file_list: list of files to check
+    :return: list of commands
+    """
+    base_cmd = "python3 -m pydocstyle --explain --source"
+
+    cmd_list = []
+    for check_file in file_list:
```

Comment:
I like more the approach with one function for all linters since we would avoid calling this `for` in each such function. Something like:
```
fn gen_linters_cmd()
cmd_for_linter1
cmd_for_linter2
for check_file in file_list:
   cmd_list.append(f"{cmd_for_linter1} {check_file}")
   cmd_list.append(f"{cmd_for_linter2} {check_file}")
```

Reply:
The problem is that not all linters check the project through similar commands. 
Some need the file name, while others only need a path, which would not make the code very clear overall.

Reply:
Indeed, I think you are referring to pycodestyle, right? Actually, seems like providing the name of the file (just like we do for pydocstyle and pylint) works for pycodestyle too. Right now we do:
```
python3 -m pycodestyle --show-pep8 --show-source '
            '--exclude=../build ../
```
But I tried
``python3 -m pycodestyle --show-pep8 --show-source ../tests/integration_tests/functional/test_api.py 
```
and it does work, so we could do that for pycodestyle too. 
This is just a suggestion. Changes look good overall.

Reply:
Fair enough, didn't check if this works. Will update it and put everything into one function.